integrations/ad_integration/tests/test_utils.py

```diff
@@ -72,3 +83,373 @@ def setUp(self):
         status = self._check_mo_ready_for_testing()
         if status:
             self.skipTest(status)
+
```

Comment:
Det er fair med en vis mængde tests. Husk dog, at der alligevel blev fundet fejl ved tests op i mod remote management serveren. 
Vi får altså ikke penge pr linie testkode :-)

Reply:
Yes, der blev fundet fejl, da vores tests ikke har særligt stor dækning, og når det sker burde vi nok udvide vores testsuite for at sikre at den fejl der blev fundet aldrig bliver reintroduceret.

Jeg er enig i at en test-suite ikke beviser fravær af fejl, men blot viser at koden virker i nogle specifikke tilfælde. Hvis vi kan vise at koden virker i mange tilfælde, og når der sker fejl, at vi kan sikre de ikke sker igen, så er vi ret langt ifht. at bevise den ønskede funktionalitet.

Vi får ikke penge pr linie testkode, vi får penge for funktionel kode der gør hvad den skal. Linjer af test-kode er med til at sikre det mål.

Reply:
Jeg er da enig et stykke hen ad vejen- og har sat tingene lidt på spidsen. Men testkoden er altså eleveret lidt over almindeligt dødelig datamatikerniveau.